File: src/Clients/accountant/src/models/entities/transaction.ts
import { Syncable } from "models/syncable";
import { Category } from "./category";

export class TransactionModel implements Syncable {
  accountName: string;
  category: Category;
  convertedAmount: number;
  synced = false;

  constructor(
    public id: number,
    public fromAccountId: number,
    public toAccountId: number,
    public categoryId: number,
    public amount: number,
    public fromStocks: number,
    public toStocks: number,
    public currency: string,
    public description: string,
    public date: string,
    public isEncrypted: boolean,
    public encryptedDescription: string,
    public salt: string,
    public nonce: string,
    public createdDate: Date,
    public modifiedDate: Date
  ) {}

  get isTax() {
    return this.category?.isTax;
  }
}

File: src/Core/Jobs/Models/Accountant/Transaction.cs
namespace Jobs.Models.Accountant;

public sealed class Transaction
{
    public int? FromAccountId { get; set; }
    public int? ToAccountId { get; set; }
    public int? CategoryId { get; set; }
    public decimal Amount { get; set; }
    public required string Currency { get; init; }
    public string? Description { get; set; }
    public DateOnly Date { get; set; }
    public bool Generated { get; set; }
    public DateTime CreatedDate { get; set; }
    public DateTime ModifiedDate { get; set; }
}
